### PR TITLE
Refactor: func in search.go

### PR DESCRIPTION
**Pull request by LimJiAn**
Hello👋
There are about three things that I modified.

Modify the position of the function
Modify variable position
Change code to readability

### CODE REVIEW

File: search.go [hunk 1]
```diff
@@ -397,20 +414,11 @@ func (r *SearchRequest) UnmarshalJSON(input []byte) error {
 	r.Score = temp.Score
 	r.SearchAfter = temp.SearchAfter
 	r.SearchBefore = temp.SearchBefore
-	r.Query, err = query.ParseQuery(temp.Q)
-	if err != nil {
+	if r.Query, err = query.ParseQuery(temp.Q); err != nil {
 		return err
 	}
 
-	if r.Size < 0 {
```
Comment (abhinavdangeti) [inline search.go:405]: I believe its in our interest to keep these checks after the query parsing to override any negative integers from the user.

File: search.go [hunk 2]
```diff
@@ -355,40 +363,49 @@ func (r *SearchRequest) SetSearchBefore(before []string) {
...
+			Sort             []json.RawMessage `json:"sort"`
+			IncludeLocations bool              `json:"includeLocations"`
+			Score            string            `json:"score"`
+			SearchAfter      []string          `json:"search_after"`
+			SearchBefore     []string          `json:"search_before"`
+		}
+		err error
+	)
+
+	if err := json.Unmarshal(input, &temp); err != nil {
```
Comment (abhinavdangeti) [inline search.go:384]: `err = json.Unmarshal`

Comment (LimJiAn) [reply search.go]: Thank you!
I understood your comment, and in order to leave the previous commit as it is, I proceeded with a forced push and modified it.

**Comment by LimJiAn** [comment]
@abhinavdangeti 

Thank you!
I understood your comment, and in order to leave the previous commit as it is, I proceeded with a forced push and modified it.

File: search.go [hunk 3]
```diff
@@ -67,13 +61,15 @@ func (dr *dateTimeRange) ParseDates(dateTimeParser analysis.DateTimeParser) (sta
 			start = s
 		}
 	}
+
```
Comment (metonymic-smokey) [inline search.go:64]: Nit: Drop this and all the other empty lines introduced in the PR.

Comment (LimJiAn) [reply search.go]: I changed it as you said.

Comment (LimJiAn) [reply search.go]: > I changed it as you said.

I changed it as you said.

Comment (LimJiAn) [reply search.go]: > Nit: Drop this and all the other empty lines introduced in the PR.

I changed it as you said.

**Review by abhinavdangeti — APPROVED** [review]
Thank you for the contribution.

**Pull request merged**